app/components/shared/gcse_qualification_cards_component.html.erb

```diff
@@ -46,10 +46,22 @@
               <dt class="app-qualification__key govuk-visually-hidden">Awarded</dt>
               <dd class="app-qualification__value app-qualification__value--caption">
               <%= qualification.award_year %>
+              <% if editable? %>
+                <%= govuk_link_to support_interface_application_form_edit_gcse_award_year_path(gcse_id: qualification.id) do %>
+                  Change <span class="govuk-visually-hidden">award year</span>
+                <% end %>
+              <% end %>
               </dd>
               <dt class="app-qualification__key">Grade</dt>
-              <% grade_details(qualification).each do |grade| %>
-                <dd class="app-qualification__value"><%= grade %></dd>
+              <% grade_details(qualification).each_with_index do |grade, index| %>
+                <dd class="app-qualification__value">
+                  <%= grade %>
+                  <% if editable? %>
+                    <%= govuk_link_to support_interface_application_form_edit_gcse_grade_path(gcse_id: qualification.id, constituent_grades: qualification.constituent_grades.present?, index: index) do %>
+                      Change <span class="govuk-visually-hidden">grade</span>
+                    <% end %>
+                  <% end %>
```

Comment:
i feel slightly uneasy about relying on array index like this. also `grade_details(qualification)` (that is, an array of `ApplicationQualificationDecorator` objects) is not the most obvious or useful thing to be iterating.

how about dealing with the 'normal' condition with an if statement (`<% if qualification.grade.present? %>`) and then for anything that's got `constituent_grades`, iterate the actual `qualification.constituent_grades` record (not `grade_details`), passing the grade key into the request params (rather than the index).

the annoying thing here is that you'd have to extract some of the logic from `ApplicationQualificationDecorator` in order to get `grade_details` text for a specific constituent grade.

what you think? maybe not worth it.